crates/nu-parser/src/parser.rs

```diff
@@ -266,13 +266,22 @@ pub fn check_name<'a>(working_set: &mut StateWorkingSet, spans: &'a [Span]) -> O
     }
 }
 
-fn parse_external_arg(working_set: &mut StateWorkingSet, span: Span) -> Expression {
+fn parse_external_arg(working_set: &mut StateWorkingSet, span: Span) -> ExternalArgument {
     let contents = working_set.get_span_contents(span);
 
     if contents.starts_with(b"$") || contents.starts_with(b"(") {
-        parse_dollar_expr(working_set, span)
+        ExternalArgument::Regular(parse_dollar_expr(working_set, span))
     } else if contents.starts_with(b"[") {
-        parse_list_expression(working_set, span, &SyntaxShape::Any)
+        ExternalArgument::Regular(parse_list_expression(working_set, span, &SyntaxShape::Any))
```

Comment:
Now that lists aren't implicitly spread, should this give a parse error so the user is warned at parse time rather than at runtime?

Reply:
Seems like it would be a good idea to me. A parse error at parse time but if they try to run it anyway it would give an error too, right?

Reply:
ya, I like the suggestion route. It's always good to do that if we know the suggestion is correct.